tests/apollo/test_skvbc_backup_restore.py

```diff
@@ -101,19 +106,128 @@ async def test_checkpoint_propagation_after_restarting_replicas(self, bft_networ
+
+        for replica in bft_network.all_replicas():
+            bv = await bft_network.get_current_view()
+            print(f" replica {replica} view before: {bv}")
+
+        # stop n replicas in a random order with a delay of 10s in between
+        stopped_replicas = await self._stop_random_replicas_with_delay(bft_network, delay=10/n)
+
+        stopped_replicas.remove(current_primary)
+
+        # start stopped replicas in a random order with a delay of 10s in between
+        # view change only happens if the initial primary starts at time > viewchangeTimer
+        # to make the test robust we start primary at the last
+        await self._start_random_replicas_with_delay(bft_network, stopped_replicas, current_primary, delay=10)
+
+        with trio.move_on_after(seconds=150):
+            while True:
+                # wait for view to stabilize
+                for replica in bft_network.all_replicas():
+                    await bft_network.wait_for_view(
+                        replica_id=replica,
+                        expected=lambda v: v > initial_view,
+                        err_msg="Make sure view is stable after all replicas are started."
+                    )
+
+        current_primary = await bft_network.get_current_primary()
+
+        await self._wait_for_processing_window_after_view_change(current_primary, bft_network)
+
+        # for vc to stabilize
+        await trio.sleep(seconds=10)
+
+        # verify checkpoint persistency
+        await bft_network.wait_for_replicas_to_checkpoint(bft_network.all_replicas(),
+                                                          expected_checkpoint_num=lambda ecn: ecn >= checkpoint_after)
+
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(),
+            num_of_checkpoints_to_add=1,
+            verify_checkpoint_persistency=False,
+            assert_state_transfer_not_started=False
+        )
+
     @staticmethod
-    async def _stop_random_replicas_with_delay(
-            bft_network, replica_set_size, delay=5, exclude_replicas=None):
-        random_replicas = bft_network.random_set_of_replicas(size=replica_set_size, without=exclude_replicas)
-        for replica in random_replicas:
+    async def _stop_random_replicas_with_delay(bft_network, delay=10, exclude_replicas=None):
+        all_replicas = bft_network.all_replicas(without=exclude_replicas)
+        random.shuffle(all_replicas)
+        for replica in all_replicas:
+            print("stopping replica: ", replica)
             bft_network.stop_replica(replica)
             await trio.sleep(delay)
-        return list(random_replicas)
+        return list(all_replicas)
 
     @staticmethod
-    async def _start_random_replicas_with_delay(
-            bft_network, stopped_replicas, delay=5):
+    async def _start_random_replicas_with_delay(bft_network, stopped_replicas, initial_primary, delay=10):
```

Comment:
same here - `delay=10`

Reply:
With lower delay, say, 5s, 2f + 1 replicas may come up in the initial view (vc_timeout is set to be 20s to give time to restarted replicas to recover), hence, not triggering a view change. If we reduce vc_timeout, say to 10s, we end up triggering unexpected view changes